pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1661,3 +1726,47 @@ func (ctrl *PersistentVolumeController) findDeletablePlugin(volume *v1.Persisten
 	}
 	return plugin, nil
 }
+
+// obtain provisioner/deleter name for a volume
+func (ctrl *PersistentVolumeController) getProvisionerNameFromVolume(volume *v1.PersistentVolume) string {
+	plugin, err := ctrl.findDeletablePlugin(volume)
+	if err != nil {
+		return "N/A"
+	}
+	// if external provisioner was used for provisioning,
+	// the volume MUST have annotation of AnnDynamicallyProvisioned, use the value
+	// as the provisioner name
+	if plugin == nil {
+		return volume.Annotations[pvutil.AnnDynamicallyProvisioned]
+	} else if plugin.IsMigratedToCSI() {
+		// in case where a plugin has been migrated to CSI,
+		// use the CSI name instead of in-tree name
+		storageClass := v1helper.GetPersistentVolumeClass(volume)
+		class, err := ctrl.classLister.Get(storageClass)
+		if err != nil {
+			return "N/A"
+		}
+		provisionerName, err := ctrl.getCSINameFromIntreeName(class.Provisioner)
+		if err != nil {
+			return "N/A"
+		}
+		return provisionerName
+	}
+	return plugin.GetPluginName()
```

Comment:
is `PluginName == ProvisionerName`? I think this is true for in-tree plugins but just want to double check

Reply:
for the last in-tree case, yes its equivalent I think